File: python/2340. Minimum Adjacent Swaps to Make a Valid Array/solution.py
class Solution:
    def minimumSwaps(self, nums: List[int]) -> int:
        minIndex = 0
        maxIndex = 0

        minNum = math.inf
        maxNum = -math.inf

        for i in range(len(nums)):
            number = nums[i]

            if number >= maxNum:
                maxNum = number
                maxIndex = i

            if number < minNum:
                minNum = number
                minIndex = i


        return (len(nums) - 1 - maxIndex) + (minIndex) - (1 if minIndex > maxIndex else 0)

File: java/2340. Minimum Adjacent Swaps to Make a Valid Array/Solution.java
class Solution {
    public int minimumSwaps(int[] nums) {
        int minIndex = 0;
        int maxIndex = 0;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] >= nums[maxIndex]) {
                maxIndex = i;
            } 
            if (nums[i] < nums[minIndex]) {
                minIndex = i;
            }
        }

        int requiredSwaps = 0;
        requiredSwaps += nums.length - 1 - maxIndex; // to move max to rightmost position
        requiredSwaps += minIndex; // to move min to leftmost position
        if (minIndex > maxIndex) requiredSwaps -= 1; // to remove duplicate swap

        return requiredSwaps;

    }
}